ray-operator/controllers/ray/raycluster_controller.go

```diff
@@ -1257,6 +1272,49 @@ func (r *RayClusterReconciler) calculateStatus(ctx context.Context, instance *ra
 			}
 		}
 
+		suspendStatus := utils.FindRayClusterSuspendStatus(newInstance)
+		if suspendStatus == rayv1.RayClusterSuspending {
+			if len(runtimePods.Items) == 0 {
+				meta.RemoveStatusCondition(&newInstance.Status.Conditions, string(rayv1.RayClusterProvisioned))
+				meta.SetStatusCondition(&newInstance.Status.Conditions, metav1.Condition{
+					Type:   string(rayv1.RayClusterSuspending),
+					Reason: string(rayv1.RayClusterSuspending),
+					Status: metav1.ConditionFalse,
+				})
+				meta.SetStatusCondition(&newInstance.Status.Conditions, metav1.Condition{
```

Comment:
We may need to consider which information in the CR status should be cleaned up. For example, https://github.com/ray-project/kuberay/blob/bee1b71e01785aaa29c6ec9a0ad6604eb868579c/ray-operator/controllers/ray/rayjob_controller.go#L314-L322

Reply:
I think the only fields we left are `HeadInfo.PodIP` and `HeadInfo.PodName`. We can update them in the `func (r *RayClusterReconciler) updateHeadInfo`:

```go
func (r *RayClusterReconciler) updateHeadInfo(ctx context.Context, instance *rayv1.RayCluster) error {
	headPod, err := common.GetRayClusterHeadPod(ctx, r, instance)
	if err != nil {
		return err
	}
	if headPod != nil {
		instance.Status.Head.PodIP = headPod.Status.PodIP
		instance.Status.Head.PodName = headPod.Name
	} else {
		instance.Status.Head.PodIP = ""   // <--- add this
		instance.Status.Head.PodName = "" // <--- add this
	}
```

I also added a test for checking whether the `Status.Head.PodName` is cleared. And `Status.Head.PodIP` is not tested because we don't have IPs in envtest.